src/savers/tvg/tvgTvgSaver.cpp

```diff
@@ -200,12 +200,59 @@ TvgBinCounter TvgSaver::serializePaint(const Paint* paint)
 }
 
 
+/* Propagate parents properties to the child so that we can skip saving the parent. */
+TvgBinCounter TvgSaver::serializeChild(const Paint* parent, const Paint* child, const Matrix* transform)
+{
+    const Paint* compTarget = nullptr;
+    auto compMethod = parent->composite(&compTarget);
+
+    //If the parent & the only child have composition, we can't skip the parent....
+    if (compMethod != CompositeMethod::None && child->composite(nullptr) != CompositeMethod::None) return 0;
+
+    //propagate opacity
+    uint32_t opacity = parent->opacity();
+
+    if (opacity < 255) {
+        uint32_t tmp = (child->opacity() * opacity);
+        if (tmp > 0) tmp /= 255;
+        const_cast<Paint*>(child)->opacity(tmp);
+    }
+
+    //propagate composition
+    if (compTarget) const_cast<Paint*>(child)->composite(unique_ptr<Paint>(compTarget->duplicate()), compMethod);
+
+    return serialize(child, transform);
+}
+
+
 TvgBinCounter TvgSaver::serializeScene(const Scene* scene, const Matrix* transform)
 {
+    auto it = this->iterator(scene);
+    if (it->count() == 0) return 0;
+
+    //Case - Only Child: Skip saving this scene.
+    if (it->count() == 1) {
```

Comment:
We may lose the matrix transformation and opacity of a scene. We should check these are default before skipping.

Reply:
here given transform is the scene's transform and opacity is to multiplied in the serializeChild() inside.
So it takes cover.

Reply:
you are right. If so, it's absolutely ok